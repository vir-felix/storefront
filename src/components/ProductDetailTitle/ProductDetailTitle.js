import React, { Component } from "react";
import PropTypes from "prop-types";
import Typography from "material-ui/Typography";
import Grid from "material-ui/Grid";

/**
 * Product detail title and pageTitle (subtitle)
 * @class ProductDetailTitle
 */
class ProductDetailTitle extends Component {
  static propTypes = {
    /**
     * Subtitle
     */
    pageTitle: PropTypes.string,

    /**
     * Main title, the h1 for the page
     */
    title: PropTypes.string
  }
  render() {
    const { title, pageTitle } = this.props;
    return (
      <Grid item sm={12}>
        {title && <Typography gutterBottom={true} variant="display2">{title}</Typography>}
        {pageTitle && <Typography color="primary" component="h2" variant="title">{pageTitle}</Typography>}
      </Grid>
    );
  }
}

export default ProductDetailTitle;
