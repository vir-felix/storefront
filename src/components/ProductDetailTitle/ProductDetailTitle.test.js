import React from "react";
import renderer from "react-test-renderer";
import ProductDetailTitle from "./ProductDetailTitle";

test("basic snapshot", () => {
  const component = renderer.create(<ProductDetailTitle title="Title" pageTitle="Subtitle" />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
