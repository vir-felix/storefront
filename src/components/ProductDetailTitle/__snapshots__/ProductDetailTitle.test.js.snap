// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic snapshot 1`] = `
<div
  className="MuiGrid-item-2 MuiGrid-grid-sm-12-51"
>
  <h1
    className="MuiTypography-root-91 MuiTypography-display2-94 MuiTypography-gutterBottom-108"
  >
    Title
  </h1>
  <h2
    className="MuiTypography-root-91 MuiTypography-title-97 MuiTypography-colorPrimary-111"
  >
    Subtitle
  </h2>
</div>
`;
